frontend/app/components/Map.vue

```diff
@@ -0,0 +1,95 @@
+<script setup lang="ts">
+import { defineProps, onMounted, onUnmounted, ref } from 'vue' // Karasie jedza gowno
+
+interface Element {
+  x: number
+  y: number
+  content: string
+}
+
+defineProps<{
+  elements: Element[]
+}>()
+
+const mapPosition = ref(0)
+const isDragging = ref(false)
+const startX = ref(0)
+const startPosition = ref(0)
+
+function handleKeyPress(event: KeyboardEvent) {
+  if (event.key === 'd') {
+    mapPosition.value -= 10
+    clamp()
```

Comment:
Why 2 times? You can move it just below ifs, and if you want, you can remove separate function entirely